### PR TITLE
Add 'ispath()' twig function, to test for an existing path / route.

### PR DESCRIPTION
**Pull request by bobdenotter**

### CODE REVIEW

**Comment by Boorj** [comment]
:bow: thanks, lightspeed coding master) :santa:

**Comment by Boorj** [comment]
Library usage is a dirty workaround, `app['routes']->get($routename)` is a proper way, i suppose (since it's used in original `UrlGenerator`).



 And, maybe, 'route_exists' or 'is_route' :panda_face:?

File: src/Twig/Handler/RecordHandler.php [hunk 1]
```diff
@@ -313,4 +314,26 @@ public function selectField($content, $fieldName, $startempty = false, $keyName
...
+     * @param bool   $relative
+     *
+     * @return bool
+     */
+    public function isPath($name, $parameters = array(), $relative = false)
+    {
+        $success = true;
+
+        try {
+            $path = Library::path($name, $parameters, $relative);
```
Comment (GawainLynch) [inline src/Twig/Handler/RecordHandler.php]: This is introducing a new dependency on a deprecated class/function.

Would be much better to use the SP directly.

File: src/Twig/Handler/RecordHandler.php [hunk 2]
```diff
@@ -313,4 +314,26 @@ public function selectField($content, $fieldName, $startempty = false, $keyName
...
+        } catch (\Exception $e) {
```
Comment (GawainLynch) [inline src/Twig/Handler/RecordHandler.php]: Also, just came back and noticed this, we should catch `\Symfony\Component\Routing\Exception\RouteNotFoundException` as a general exception could point to other problems in the stack that would be silently buried.

**Pull request merged**